### ISSUE TITLE
put maximum recursion depth to module.pyrpl - property

### ISSUE THREAD

**Issue by lneuhaus** [lneuhaus/pyrpl#340]
Issue comes from here: https://github.com/RedPitaya/RedPitaya/issues/150

Copy-paste: 
Forgive me if my format is terrible, I do not usually file issues.

Whenever I call/assign any member of redpitaya.scope, my code hangs forever and I have traced it back to the pyrpl/modules.py in the pyrpl function. I do not have a good enough understanding of the code to make a suggestion but could this be a bug?

```
def pyrpl(self):
    """
    Recursively looks through patent modules untill pyrpl instance is
    reached.
    """
    from .pyrpl import Pyrpl
    parent = self.parent
    while (not isinstance(parent, Pyrpl)):
        parent = parent.parent
    return parent
```

**Comment by lneuhaus** [comment]
I agree this is while-loop bit smelly, so i suggest we implement a maximum recursion depth of 100 or so here. 

Nevertheless, I would like to understand how there can be an infinite loop here. 

The depth of any parent/child chain is necessarily finite, unless something is its own parent or the parent of something that appeared already in the chain. 

Could you please provide some extra context so we can try to debug this? I.e.:
- did you modify the code, and how?
- how did you get to this point, i.e. did you run pyrpl from the code of the current master branch of pyrpl, and on what python version are you working?
- or simply hit Ctrl-C in a notebook/ipython console when this infinite loop occurs, type %debug, and scroll up and display the value of parent at each stage, to reconstruct what the parents of scope actually are - i would expect something like scope, redpitaya, scope, redpitaya, ..., which would indicate that scope is a parent of redpitaya, which would raise the question where this hypothetical setting comes from..

Thanks in advance!

PS: If you have not played with the source code, I rather do not believe that the issue you describe is the reason for your problem, but of course we have to exclude this possibility first to be able to find the actual problem.

**Comment by lneuhaus** [comment]
fixed in afa561ddde32df399a13b0a71449ee0447cad1e8 (currently only on branch develop-0.9.3)
